From 719e1874ecc84a5b0b5a1ff411c7fe7044790160 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 23 Mar 2018 08:23:11 -0500
Subject: [PATCH 3539/5242] MLK-17703-6: drm: imx: dcss: fix output
 colorimetry in crtc

commit  bc487b9c2d577e393935bc85895735643174382c from
https://source.codeaurora.org/external/imx/linux-imx.git

The detection of the supported output colorimetry was wrong. This patch will
fix that and, also, get rid of the REC2100HLG EOTF setting for now. It produces
bad colors.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-crtc.c |   12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-crtc.c b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
index bff5dab..b7eeade 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-crtc.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
@@ -171,19 +171,17 @@ void dcss_crtc_setup_opipe(struct drm_crtc *crtc, struct drm_connector *conn,
 	struct drm_display_info *di = &conn->display_info;
 	int vic;
 
-	if ((colorimetry & HDMI_EXTENDED_COLORIMETRY_BT2020) ||
-	    (colorimetry & HDMI_EXTENDED_COLORIMETRY_BT2020_CONST_LUM))
+	if ((colorimetry & BIT(HDMI_EXTENDED_COLORIMETRY_BT2020)) ||
+	    (colorimetry & BIT(HDMI_EXTENDED_COLORIMETRY_BT2020_CONST_LUM)))
 		dcss_crtc->opipe_g = G_REC2020;
-	else if (colorimetry & HDMI_EXTENDED_COLORIMETRY_ADOBE_RGB)
+	else if (colorimetry & BIT(HDMI_EXTENDED_COLORIMETRY_ADOBE_RGB))
 		dcss_crtc->opipe_g = G_ADOBE_ARGB;
-	else if (colorimetry & HDMI_EXTENDED_COLORIMETRY_XV_YCC_709)
+	else if (colorimetry & BIT(HDMI_EXTENDED_COLORIMETRY_XV_YCC_709))
 		dcss_crtc->opipe_g = G_REC709;
 	else
 		dcss_crtc->opipe_g = G_REC601_PAL;
 
-	if (eotf & (1 << 3))
-		dcss_crtc->opipe_nl = NL_2100HLG;
-	else if (eotf & (1 << 2))
+	if (eotf & (1 << 2))
 		dcss_crtc->opipe_nl = NL_REC2084;
 	else
 		dcss_crtc->opipe_nl = NL_REC709;
-- 
1.7.9.5

